pymatgen/analysis/quasirrho.py

```diff
@@ -0,0 +1,262 @@
+        """
+        Calculate Quasi-RRHO thermochemistry
+        Args:
+            mol (Molecule): Pymatgen molecule
+            mult (int): Spin multiplicity
+            sigma_r (int): Rotational symmetry number
+            frequencies (list): List of frequencies [cm^-1]
+            elec_energy (float): Electronic energy [Ha]
+        """
+        # Calculate mass in kg
+        mass = 0
+        for site in mol.sites:
+            mass += site.specie.atomic_mass
+        mass *= amu_to_kg
+
+        # Calculate vibrational temperatures
+        vib_temps = []
+        for f in frequencies:
+            if f > 0:
+                vib_temps.append(f * c * h / kb)
+
+        # Translational component of entropy and energy
+        qt = (2 * np.pi * mass * kb * self.temp / (h * h)) ** (3 / 2) * kb * self.temp / self.press
+        st = R * (np.log(qt) + 5 / 2)
+        et = 3 * R * self.temp / 2
+
+        # Electronic component of Entropy
+        se = R * np.log(mult)
+
+        # Get properties related to rotational symmetry. Bav is average moment of inertia
+        Bav, i_eigen = get_avg_mom_inertia(mol)
+
+        # Check if linear
+        coords = mol.cart_coords
+        v0 = coords[1] - coords[0]
+        linear = True
+        for coord in coords[1:]:
+            theta = abs(np.dot(coord - coords[0], v0) / np.linalg.norm(coord - coords[0]) / np.linalg.norm(v0))
+            if not isclose(theta, 1, abs_tol=1e-4):
+                linear = False
+
+        # Rotational component of Entropy and Energy
+        if linear:
+            i = np.amax(i_eigen)
+            qr = 8 * np.pi**2 * i * kb * self.temp / (sigma_r * (h * h))
+            sr = R * (np.log(qr) + 1)
+            er = R * self.temp
+        else:
+            rot_temps = [h**2 / (np.pi**2 * kb * 8 * i) for i in i_eigen]
+            qr = np.sqrt(np.pi) / sigma_r * self.temp ** (3 / 2) / np.sqrt(rot_temps[0] * rot_temps[1] * rot_temps[2])
+            sr = R * (np.log(qr) + 3 / 2)
+            er = 3 * R * self.temp / 2
+
+        # Vibrational component of Entropy and Energy
+        ev = 0
+        sv_quasiRRHO = 0
+        sv = 0
+
+        for vt in vib_temps:
+            ev += vt * (1 / 2 + 1 / (np.exp(vt / self.temp) - 1))
+            sv_temp = vt / (self.temp * (np.exp(vt / self.temp) - 1)) - np.log(1 - np.exp(-vt / self.temp))
+            sv += sv_temp
+
+            mu = h / (8 * np.pi**2 * vt * c)
+            mu_prime = mu * Bav / (mu + Bav)
+            srotor = 1 / 2 + np.log(np.sqrt(8 * np.pi**3 * mu_prime * kb * self.temp / h**2))
+            weight = 1 / (1 + (self.v0 / vt) ** 4)
+            sv_quasiRRHO += weight * sv_temp + (1 - weight) * srotor
+
+        sv_quasiRRHO *= R
+        sv *= R
+        ev *= R
+        etot = (et + er + ev) * kcal2hartree / 1000
+        self.h_corrected = etot + R * self.temp * kcal2hartree / 1000
+        molarity_corr = R_ha * self.temp * np.log(R_volume * self.temp * self.conc)
```

Comment:
It looks like the molarity correction is `RT log (RTC)`. Shouldn't it just be `RT log(C`? I may be ignorant, but please double check / confirm.

Reply:
Also, this formulation implicitly assumes a standard state of 1 M. It might be good to state that in the docstring in case someone is using a calc with a different standard state

Reply:
I was following Wheeler's script for the concentration correction. I believe it comes from a thermodynamic cycle for cluster continuum models that leads to an extra `RT log(R_volume T)`. It's equations 7 and 8 in the [Ho and Coote pKa paper](https://link.springer.com/article/10.1007/s00214-009-0667-0). 
Thanks for the point about the 1M assumption. I will add that.

Reply:
Ah, thanks for clarifying. So looking at that section of the paper:
![image](https://github.com/materialsproject/pymatgen/assets/1908695/99fdbdbd-e453-4f32-bb5d-6112a2a8e9e5)

I'm pretty sure the extra `-RT log [R~T]` term is there to adjust the reference states (the paragraph above and the different units of R and R~ are the clues). The solvation energy assumes 1 mol/L in both gas and solution, so you need a correction for that difference. The numerical value of R~T is 24.43 atm / (mol/L), which is the same value one might use to make a standard state correction.

So I think it should be `RT ln C`, and I'm glad we have the standard state in the docstring to hopefully call attention to the need to be careful interpreting the number. Does that make sense?

Reply:
Your explanation of the source of the extra term makes sense to me, but I don't understand why we shouldn't include the standard state correction when providing a concentration-corrected free energy in solvent. If RTln(concentration) uses mol/L, shouldn't we also correct for the 1 atm  standard state used in the DFT code?

Reply:
That's a fair point. To me what makes it tricky is that you only need to do the standard state correction when you are writing a _reaction energy_ between species with different reference states, but this module just computes a single correction for a single species. (in other words, the reason for the correction is to make sure all species use the same reference state). We don't necessarily know that users of this class are going to take this free energy and combine it with a gas phase energy.

The other small issue is that I think not every DFT code necessarily uses the same reference state. But I think both Q-Chem and Gaussian use 1 atm for gas phase calcs, and as long as we document what we're doing I'm not too worried about that.

Personally I think it will be more transparent and less confusing to users to just do RT ln C and let them adjust as needed if they have to change reference states. But again, as long as we document what's going on it could be OK to build it in.

If you want to build it in, I suggest writing it as `( RT log ( C) - RT log (1/24.55) )` and adding a comment line to make explicitly clear that you're dividing C by the molarity of an ideal gas at 1 atm.

Do others like @samblau or @espottesmith or @materialsproject/second-foundation have thoughts?

Reply:
I definitely agree that we should be more specific in the documentation, and I'll put that in. Maybe with a reference to describe it in detail if I can find a really good one. I also think of the free energy in implicit solvent as already containing an assumed change in free energy from vacuum to solution, and would thus include the standard state correction when using implicit solvent. Is that correct?

I would also appreciate input from everyone you tagged! 

I think another option would be to just entirely remove the concentration correction for now. It is somewhat just a legacy from converting a script that was useful to me into Pymatgen.

Reply:
> I definitely agree that we should be more specific in the documentation, and I'll put that in. Maybe with a reference to describe it in detail if I can find a really good one. I also think of the free energy in implicit solvent as already containing an assumed change in free energy from vacuum to solution, and would thus include the standard state correction when using implicit solvent. Is that correct?

I just think of the free energy of an isolated molecule as "DFT energy (~ 0K enthalpy) plus T-dependent enthalpy and entropy".

My understanding is that implicit solvent models adjust the free energy for _interactions between the solute and the medium_ but do not account for changes in standard state (vac -> solution). Unless we're talking about _formation free energy_ or a _reaction energy_, the free energy of an isolated molecule begs the question "relative to what"? And you don't have to introduce a standard/reference state until you answer that question. I know some solvent models output a solvation (reaction) energy, but I believe that implicitly assumes the same reference state for both vacuum and solvent. There's nothing inherently wrong with that but it's often not what the end user needs for comparison to experiment.

> 
> I think another option would be to just entirely remove the concentration correction for now. It is somewhat just a legacy from converting a script that was useful to me into Pymatgen.

Yeah, the more I think about it, that might be the cleanest thing to do here. This module is for calculating QRRHO thermochemistry, which (unless I'm mistaken) does not _specifically_ have anything to do with an implicit solvent model and does not make any assumptions about the concentration; it's just a way of dampening low-frequency contributions to rot and vib entropy. So in the same way that standard thermochemistry output in Gaussian/Q-Chem doesn't make any adjustments for concentration, maybe this shouldn't either.

Reply:
I'm still not entirely convinced, mainly because implicit solvent models are made to capture solvation free energies, but I definitely understand what you're saying. I'd love to have this conversation offline more too, given how fundamentally important it is to get these corrections correct, but I agree that that can be beyond the scope of this PR. I'll remove it for now and maybe if Pymatgen wants to start recommending free energy treatments we can implement more general thermochemistry!

Reply:
Happy to discuss offline!